controller/api/destination/endpoint_translator.go

```diff
@@ -102,9 +116,11 @@ func (et *endpointTranslator) sendFilteredUpdate(set watcher.AddressSet) {
 
 	if len(diffAdd.Addresses) > 0 {
 		et.sendClientAdd(diffAdd)
+		et.watchEndpointPolicy(diffAdd)
```

Comment:
we probably need locking around watchEndpointPolicy and  closeEndpointPolicy to avoid race conditions

Reply:
refer to this [comment](https://github.com/linkerd/linkerd2/pull/6889#discussion_r712674796)